### PR TITLE
JBPM-5573: Stunner - Integration of latest KIE Forms API & implementations

### PR DESCRIPTION
**Pull request by pefernan**
- Added new Annotation Driven Forms engine.
  - Added new Annotations
  - Added Processos for source generation
  - Added rendering engine for the new ADF engine
- Addapted Stunner forms to the new ADF engine

These PR's should be merged together:
https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/364
https://github.com/droolsjbpm/kie-wb-common/pull/640
https://github.com/droolsjbpm/jbpm-console-ng/pull/609
https://github.com/droolsjbpm/jbpm-designer/pull/533
https://github.com/droolsjbpm/appformer/pull/66
https://github.com/droolsjbpm/kie-wb-distributions/pull/446

### CODE REVIEW

**Comment by pefernan** [comment]
@wmedvede can you take a look... there are millions of changes but I can help you.

**Comment by pefernan** [comment]
@romartin can you take a look... there are a lot of changes but I can help.

**Comment by romartin** [comment]
Hi @pefernan , I'm not able to run the Stunner's showcase as it throws an error running the container, so on server side, about some missing UF social stuff. 
On the other hand, @hasys realized that the code style is not the latest one for our KIE conventions, can you please review this too? 
Will give it a try tomorrow again. Thanks!

**Comment by hasys** [comment]
Hi @pefernan, ping me (kgaevski) if you need any help with latest code style settings. It can work (almost) automatically after some easy configurations.

**Comment by pefernan** [comment]
jenkins retest this.

**Comment by pefernan** [comment]
@romartin @hasys just fixed the code style issue. 

@romartin also fixed a dependency issue on the stunner-showcase-project related with the social-activities... a undesired provided scope ;)

File: kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-project/pom.xml [hunk 1]
```diff
@@ -223,7 +223,6 @@
     <dependency>
       <groupId>org.uberfire</groupId>
       <artifactId>uberfire-social-activities-api</artifactId>
-      <scope>provided</scope>
```
Comment (manstis) [inline kie-wb-common-stunner/kie-wb-common-stunner-showcase/kie-wb-common-stunner-showcase-project/pom.xml]: This has already been fixed on master (or so I believe it was for this module).

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-processors/src/main/java/org/kie/workbench/common/stunner/core/processors/rule/DockingRuleGenerator.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public StringBuffer generate(final String packageName,
                                  final ProcessingEnvironment processingEnvironment) throws GenerationException {
         final Messager messager = processingEnvironment.getMessager();
         messager.printMessage(Diagnostic.Kind.NOTE,
-                              "Starting code generation for [" + className + "]");
+                              "Starting code adf for [" + className + "]");
```
Comment (pefernan) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-processors/src/main/java/org/kie/workbench/common/stunner/core/processors/rule/DockingRuleGenerator.java]: @romartin it is a typo...

**Comment by pefernan** [comment]
@romartin fixed

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/forms/model/AssigneeEditorFieldDefinition.java [hunk 3]
```diff
@@ -18,49 +18,42 @@
...
-import static org.kie.workbench.common.stunner.bpmn.util.FieldLabelConstants.FIELDDEF_DEFAULT_VALUE;
-import static org.kie.workbench.common.stunner.bpmn.util.FieldLabelConstants.FIELDDEF_TYPE;
 
 @Portable
 @Bindable
-public class AssigneeEditorFieldDefinition extends FieldDefinition {
+public class AssigneeEditorFieldDefinition extends AbstractFieldDefinition<AssigneeEditorFieldType> {
 
-    public static final String CODE = "AssigneeEditor";
+    public static AssigneeEditorFieldType FIELD_TYPE = new AssigneeEditorFieldType();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/forms/model/AssigneeEditorFieldDefinition.java]: Hi @pefernan, maybe "final" is missing here.

**Comment by pefernan** [comment]
@hasys done thanks!

**Comment by pefernan** [comment]
@romartin can you review it again?

**Comment by csadilek** [comment]
jenkins retest this

**Pull request merged**